cmd/telescopes/main.go

```diff
@@ -85,17 +84,14 @@ func main() {
 
 	ciCli := recommender.NewCloudInfoClient(pc)
 
-	vmSelector := vms.NewVmSelector(logger, ciCli)
-	nodePoolSelector := nodepools.NewNodePoolSelector(logger, vmSelector)
-
-	engine := recommender.NewEngine(nodePoolSelector, logger)
-
 	// configure the gin validator
 	err = api.ConfigureValidator(ciCli)
 	emperror.Panic(err)
 
+	eng := engine.NewEngine(logger, ciCli)
+
 	buildInfo := buildinfo.New(Version, CommitHash, BuildDate)
-	routeHandler := api.NewRouteHandler(engine, buildInfo, ciCli, logger)
+	routeHandler := api.NewRouteHandler(eng, buildInfo, ciCli, logger)
```

Comment:
remove the "ciCli" field of the routhandler component, use the client of the engine where necessary. Also use the ...Source interface in the validator